packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -0,0 +1,657 @@
+  style: BorderStyle.solid,
+  width: 0.0,
+);
+const Border _kDefaultRoundedBorder = Border(
+  top: _kDefaultRoundedBorderSide,
+  bottom: _kDefaultRoundedBorderSide,
+  left: _kDefaultRoundedBorderSide,
+  right: _kDefaultRoundedBorderSide,
+);
+// Counted manually on magnified simulator.
+const BoxDecoration _kDefaultRoundedBorderDecoration = BoxDecoration(
+  border: _kDefaultRoundedBorder,
+  borderRadius: BorderRadius.all(Radius.circular(4.0)),
+);
+
+// Default iOS style from HIG specs with larger font.
+const TextStyle _kDefaultTextStyle = TextStyle(
+  fontFamily: '.SF Pro Text',
+  fontSize: 17.0,
+  letterSpacing: -0.38,
+  color: CupertinoColors.black,
+  decoration: TextDecoration.none,
+);
+
+// Value extracted via color reader from iOS simulator.
+const Color _kSelectionHighlightColor = Color(0x667FAACF);
+const Color _kInactiveTextColor = Color(0xFFC2C2C2);
+const Color _kDisabledBackground = Color(0xFFFAFAFA);
+
+/// Visibility of text field overlays based on the state of the current text entry.
+///
+/// Used to toggle the visibility behavior of the optional decorating widgets
+/// surrounding the [EditableText] such as the clear text button.
+enum OverlayVisibilityMode {
+  /// Overlay will never appear regardless of the text entry state.
+  never,
+  /// Overlay will only appear when the current text entry is not empty.
+  ///
+  /// This includes pre-filled text that the user did not type in manually. But
+  /// does not include text in placeholders.
+  editing,
+  /// Overlay will only appear when the current text entry is empty.
+  ///
+  /// This also includes not having pre-filled text that the user did not type
+  /// in manually. Texts in placeholders are ignored.
+  notEditing,
+  /// Always show the overlay regardless of the text entry state.
+  always,
+}
+
+/// An iOS-style text field.
+///
+/// A text field lets the user enter text, either with a hardware keyboard or with
+/// an onscreen keyboard.
+///
+/// This widget corresponds to both a [UITextField] and an editable [UITextView]
+/// on iOS.
+///
+/// The text field calls the [onChanged] callback whenever the user changes the
+/// text in the field. If the user indicates that they are done typing in the
+/// field (e.g., by pressing a button on the soft keyboard), the text field
+/// calls the [onSubmitted] callback.
+///
+/// To control the text that is displayed in the text field, use the
+/// [controller]. For example, to set the initial value of the text field, use
+/// a [controller] that already contains some text such as:
+///
+/// ```dart
+/// class _MyWidgetState extends State<MyWidget> {
+///   TextEditingController _textController;
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     _textController = TextEditingController(text: 'initial text');
+///   }
+///
+///   @override
+///   Widget build(BuildContext context) {
+///     return CupertinoTextField(controller: _textController);
+///   }
+/// }
+/// ```
```

Comment:
sample code should be in a `## Sample code` section so it gets analyzed.

Reply:
done